crates/typst-syntax/src/lexer.rs

```diff
@@ -182,6 +224,197 @@ impl Lexer<'_> {
+                    // be whitespaces until the line ends.
+                    self.s.eat_until(char::is_whitespace);
+                    self.error("unexpected characters after end of annotation")
+                }
+                Some(c) if is_id_start(c) => {
+                    self.s.eat_while(is_id_continue);
+                    SyntaxKind::Ident
+                }
+                Some('"') => self.annotation_string(),
+                Some(')') if has_opening_paren => {
+                    found_closing_paren = true;
+                    SyntaxKind::RightParen
+                }
+                // Explicitly detect comments for more helpful errors
+                Some('/') if self.s.at(['/', '*']) => {
+                    if self.s.eat() == Some('*') {
+                        // Found a block comment. Advance until the next
+                        // newline or '*/' just for a more accurate error span.
+                        while !self.s.eat_if("*/") && !self.s.at(is_newline) {
+                            self.s.eat();
+                        }
+                    } else {
+                        self.s.eat_until(is_newline);
+                    }
+                    self.error(eco_format!("unexpected comment inside annotation"))
+                }
+                Some(_) => {
+                    self.s.eat_until(|c: char| {
+                        c.is_whitespace() || has_opening_paren && c == ')'
+                    });
+                    self.error(eco_format!(
+                        "expected identifier{} in annotation",
+                        if has_opening_paren {
+                            ", string or closing paren"
+                        } else {
+                            " or string"
+                        }
+                    ))
+                }
+                None => break,
+            };
+
+            let node = self.emit_token(token, current_start);
+            subtree.push(node);
+        }
+
+        // Right parenthesis (covered above)
+        if has_opening_paren && !found_closing_paren {
+            subtree.push(
+                self.emit_error(
+                    "expected closing paren after annotation",
+                    self.s.cursor(),
+                ),
+            );
+        }
+
+        SyntaxNode::inner(SyntaxKind::Annotation, subtree)
+    }
+
+    /// Lexes an annotation name.
+    ///
+    /// An annotation name is an identifier within a specific subset of allowed
+    /// identifiers. Currently, `allow` is the only valid annotation name.
+    fn annotation_name(&mut self, start: usize) -> SyntaxKind {
+        self.s.eat_while(is_id_continue);
+        let ident = self.s.from(start);
+
+        if ident == "allow" {
+            SyntaxKind::AnnotationName
+        } else {
+            let error = self.error(eco_format!("invalid annotation name"));
+            self.hint("must be 'allow'");
+            error
```

Comment:
```suggestion
            self.error(eco_format!("invalid annotation name"))
                .hint("must be 'allow'")
```
Not sure if it works like this